mne/io/edf/edf.py

```diff
@@ -311,92 +334,479 @@ def _get_edf_info(fname, stim_channel, annot, annotmap, eog, misc, exclude,
+                physical_max = np.array([np.fromstring(fid.read(8), \
+                    np.float64)[0] for ch in channels])
+                digital_min = np.array([np.fromstring(fid.read(8), \
+                    np.int64)[0] for ch in channels])
+                digital_max = np.array([np.fromstring(fid.read(8), \
+                    np.int64)[0] for ch in channels])
+                prefiltering = [fid.read(80).strip().decode()
+                                for ch in channels][:-1]
+                highpass = np.ravel([re.findall('HP:\s+(\w+)', filt)
+                                     for filt in prefiltering])
+                lowpass = np.ravel([re.findall('LP:\s+(\w+)', filt)
+                                    for filt in prefiltering])
+
+                edf_info['physical_min'] = physical_min
+                edf_info['digital_min'] = digital_min
+
+                # number of samples per record
+                n_samps = np.array([np.fromstring(fid.read(4), np.int32)[0]
+                                    for ch in channels])
+                edf_info['n_samps'] = n_samps
+
+                gdftype = np.array([np.fromstring(fid.read(4), np.int32)[0]
+                                    for ch in channels])
+                edf_info['gdf_type'] = gdftype
+                edf_info['gdf_type_np'] = [__GDFTYP_NAME[t] for t in gdftype]
+
+                fid.read(32 * edf_info['nchan']).decode()  # reserved
+                assert fid.tell() == header_nbytes
+
+                # total number of byter for data
+                edf_info['bytes_tot'] = np.sum([__GDFTYP_BYTE[t] * n_samps[i] \
+                    for i, t in enumerate(edf_info['gdf_type'])])
+
+                # EVENT TABLE
+                etp = header_nbytes + n_records * edf_info['bytes_tot']
+                # skip data to go to event table
+                fid.seek(etp)
+                etmode = fid.read(1)
+                if etmode != '':
+                    etmode = np.fromstring(etmode, np.uint8).tolist()[0]
+                    sr = np.fromstring(fid.read(3), np.uint8)
+                    event_sr = sr[0]
+                    events = []
+                    for i in range(1, len(sr)):
+                        event_sr = event_sr + sr[i] * 256 ** i
+                    N = np.fromstring(fid.read(4), np.uint32).tolist()[0]
+                    POS = np.fromstring(fid.read(N * 4), np.uint32)
+                    TYP = np.fromstring(fid.read(N * 2), np.uint16)
+
+                    if etmode == 3:
+                        CHN = np.fromstring(fid.read(N * 2), np.uint16)
+                        DUR = np.fromstring(fid.read(N * 4), np.uint32)
+                        events.append([N, POS, TYP, CHN, DUR])
+                    else:
+                        DUR = np.zeros(N, dtype=np.uint32)
+                        events.append([N, POS, TYP])
+
+                    edf_info['events'] = events
+                    edf_info['events'] = np.c_[POS, DUR, TYP]
+
```

Comment:
Instead of burying these in very long conditionals, it would be nicer to use matched-API private functions like `edf_info = _read_bdf_info(...)`, `edf_info = _read_edf_info(...)` or whatever the appropriate API would be.